magmi/inc/magmi_engine.php

```diff
@@ -513,7 +513,26 @@ public function connectToMagento()
                 $user = $default_setup->username;
                 $pass = $default_setup->password;
                 $port = $default_setup->port;
-            } else {
+            }else if($conn == 'envphp'){ //berto - scraping env.php
+				$baseDir = $this->getProp('MAGENTO', 'basedir');
+                //$envPath = $baseDir.'/app/etc/env.php';
+                $envPath = dirname(__FILE__)."/../../app/etc/env.php";
+                //$envPath = $baseDir.'/app/etc/env.php';
+                if (!file_exists($envPath)) {
+                    throw new Exception("Cannot load xml from path '$envPath'");
+                }
+                //$env_array = include dirname(__FILE__)."/../../app/etc/env.php";
+				$env_array = include $envPath;
+				//$host = $GLOBALS['env_array']['db']['connection']['default']['host'];
+				$host = $env_array['db']['connection']['default']['host'];
+                //$dbname = $GLOBALS['env_array']['db']['connection']['default']['dbname'];
+                $dbname = $env_array['db']['connection']['default']['dbname'];
+                //$user = $GLOBALS['env_array']['db']['connection']['default']['username'];
+                $user = $env_array['db']['connection']['default']['username'];
+                //$pass = $GLOBALS['env_array']['db']['connection']['default']['password'];
+                $pass = $env_array['db']['connection']['default']['password'];
+                $port = '3306';
```

Comment:
can't port also be passed as env value?

Reply:
In the env.php file I used for testing, there isn't a "port" value. So I forced the default one.

Reply:
Then please check if its set and apply default value only if its not present.